src/Microsoft.AspNet.Mvc.Extensions/JsonResult.cs

```diff
@@ -86,6 +87,16 @@ public override Task ExecuteResultAsync([NotNull] ActionContext context)
                 response.StatusCode = StatusCode.Value;
             }
 
+            if (_serializerSettings == null)
```

Comment:
I don't really mind too much because these result types are single-use objects, but it's cleaner to avoid mutating the state of the object and just use a local variable. 

As a general principle, it makes debugger harder if you have to think about *what's changed* after calling a method.